medcat/meta_cat.py

```diff
@@ -212,32 +228,52 @@ def train_raw(self, data_loaded: Dict, save_dir_path: Optional[str] = None) -> D
                                  replace_center=g_config['replace_center'], prerequisites=t_config['prerequisites'],
                                  lowercase=g_config['lowercase'])
 
-        # Check is the name there
+        # Check is the name present
         category_name = g_config['category_name']
         if category_name not in data:
             raise Exception(
                 "The category name does not exist in this json file. You've provided '{}', while the possible options are: {}".format(
                     category_name, " | ".join(list(data.keys()))))
 
         data = data[category_name]
+        if data_:
```

Comment:
you're running tokenzier here on the `sample[0]`, but in `prepare_from_json`, the tokenzier is passed in there and ran for the LSTM model.

Can we reuse prepare_from_json ?

Reply:
`self.tokenizer` is used for both:

- `prepare_from_json` which processes the actual data from the medcat export (used by bert and lstm)
-  processing `data_` (oversampled data)

I can try and re-use `prepare_from_json`, however the structure for the oversampled data would need to match the medcat export one (which is complex) and would entail re-processing some bits (calculating cpos and others)

Reply:
Re-structured to include this in data_utils under `prepare_for_oversampled_data` function